atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractTransactionManager.java

```diff
@@ -82,7 +84,9 @@ protected final <T, E extends Exception> T runTaskThrowOnConflictWithCallback(
         } finally {
             callback.run();
```

Comment:
As this is written, shouldn't we almost wrap this in a closer? like callback could throw? would we not leak things potentially then?